.pre-commit-config.yaml

```diff
@@ -0,0 +1,47 @@
+# hint: pre-commit run --all-files
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+      # exclude_types: [jupyter]
+      - id: check-yaml
+      - id: check-toml
+      - id: check-ast
+      # - id: check-docstring-first  # let variables have docstrings
+      - id: check-merge-conflict
+      - id: check-added-large-files
+      - id: mixed-line-ending
+        exclude: |
+          apstools/migration/config.*
+          LICENSE.txt
+      - id: end-of-file-fixer
+        exclude: |
+          apstools/migration/config.*
+          LICENSE.txt
+      - id: trailing-whitespace
+        exclude: |
+          apstools/migration/config.*
+          LICENSE.txt
+      - id: check-case-conflict
+      # - id: check-json
+      # - id: check-symlinks
+      - id: check-executables-have-shebangs
+  # Don't strip output from jupyter notebooks ATM.
+  # - repo: https://github.com/kynan/nbstripout
+  #   rev: 0.7.1
+  #   hooks:
+  #     - id: nbstripout
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.4.8
+    hooks:
+      - id: ruff # replaces Flake8, isort, pydocstyle, pyupgrade
+        args:
+          - --fix
+        exclude: |
```

Comment:
What about adding `apstools/utils/tests/data/*` to the exclude list?  No need to lint/format/touch any of that.  Also apply to other data files in future PRs.